modules/gapi/CMakeLists.txt

```diff
@@ -250,4 +277,10 @@ if(HAVE_ONNX)
 endif()
 
 ocv_add_perf_tests()
-ocv_add_samples()
+if(HAVE_ONEVPL)
+    #TODO Why is it? CMake `PUBLIC` dependency must be carried to dependent target of ${the_module}?
```

Comment:
I don't think it's a good idea to make it `PUBLIC`. Actually, I don't see any `PUBLIC` linkage/definitions in G-API except for standalone mode